### PR TITLE
Library: Use new columns

### PR DESCRIPTION
**Pull request by erri120**
Due to the composition aspect of the new system, there is quite the amount of boilerplate code.

TODO:

- update style selectors

### CODE REVIEW

File: src/Abstractions/NexusMods.Abstractions.NexusWebApi/Types/V2/GameId.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public static GameId FromGameDomain(GameDomain domain)
     {
         return domain.Value switch
         {
-            "stardewvalley" => (GameId)1704,
+            "stardewvalley" => (GameId)1303,
```
Comment (Al12rs) [inline src/Abstractions/NexusMods.Abstractions.NexusWebApi/Types/V2/GameId.cs:26]: What is this about?

Comment (Sewer56) [reply src/Abstractions/NexusMods.Abstractions.NexusWebApi/Types/V2/GameId.cs:26]: It was an accident when I introduced the temporary mapping, during copy-pasting. My branch which throws out as much V1 as possible does proper resolution via the website.

I noticed this bug this morning, and @erri120 ran into it later too.

File: src/NexusMods.App.UI/Pages/LibraryPage/LocalFileLibraryItemModel.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            FormattedSize,
+            DownloadedDate,
+            FormattedDownloadedDate,
+            InstalledDate,
+            FormattedInstalledDate,
+            InstallItemCommand,
+            IsInstalled,
+            InstallButtonText
+        );
+    }
```
Comment (Al12rs) [inline src/NexusMods.App.UI/Pages/LibraryPage/LocalFileLibraryItemModel.cs:57]: I hate we have to do this. 
It seems way to easy to just forget something, or not add it here when something new is added.

Comment (erri120) [reply src/NexusMods.App.UI/Pages/LibraryPage/LocalFileLibraryItemModel.cs:57]: Not much you can do about that.

Comment (Sewer56) [reply src/NexusMods.App.UI/Pages/LibraryPage/LocalFileLibraryItemModel.cs:57]: :(

Comment (Al12rs) [reply src/NexusMods.App.UI/Pages/LibraryPage/LocalFileLibraryItemModel.cs:57]: :(

File: src/NexusMods.App.UI/Pages/LibraryPage/LibraryViewModel.cs [hunk 3]
```diff
@@ -167,10 +168,19 @@ public LibraryViewModel(
...
+                            var libraryItem = LibraryItem.Load(_connection.Db, id);
+                            if (!libraryItem.IsValid()) continue;
+                            await InstallLibraryItem(libraryItem, _loadout, cancellationToken);
+                        }
+                    }
+                    else
+                    {
+                        var libraryItem = LibraryItem.Load(_connection.Db, singleId);
+                        if (!libraryItem.IsValid()) return;
                         await InstallLibraryItem(libraryItem, _loadout, cancellationToken);
```
Comment (Al12rs) [inline src/NexusMods.App.UI/Pages/LibraryPage/LibraryViewModel.cs:184]: Was there no way to just always pass a collection, and just pas a collection of one item in the cases of a single item? 
Or is there an advantage of keeping the cases separate?

Comment (erri120) [reply src/NexusMods.App.UI/Pages/LibraryPage/LibraryViewModel.cs:184]: It just looked weird to return an array with a single item. Now that we're using composition, we can have these distinctions.

**Review by Al12rs — COMMENTED** [review]
Code looks good, I ran into the
`[ERROR] While commiting transaction|System.InvalidOperationException: Invalid UTF-8 sequence.`
bug when testing mass deletions etc, that should be solved by #2154.

Would like to get that in first so this can be tested better.

**Review by Al12rs — APPROVED** [review]
I tested now with the fix from #2154 and it appears to be working.
Another pr seems to have been merged in the meantime causing another merge conflict but this PR should be good

**Review by Al12rs — APPROVED** [review]

**Pull request merged**